working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1374,12 +1467,15 @@ The context type schema for a pattern `p` is:
     var [[int x]!] = [[]]; // Infers List<List<int>?> for the list literal.
     ```
 
-*   **Literal** or **constant**: The context type schema is the static type of
-    the pattern's constant value expression.
+*   **Constant**: The context type schema is the static type of the pattern's
+    value.
 
 *   **Variable**:
 
-    1.  If `p` has no type annotation, the context type schema is `?`.
+    1.  In an assignment context, the context type schema is the static type of
+        the variable that `p` resolves to.
```

Comment:
(Should we disallow assigning the same variable more than once in the same assignment pattern?)

Reply:
Yes, that's covered in the initial syntax section for pattern assignments.